backends/tc/backend.cpp

```diff
@@ -122,15 +121,34 @@ void Backend::serialize() const {
             outstream->flush();
         }
     }
+    auto progName = options.file;
+    auto filename = progName.findlast('/');
+    if (filename) progName = filename;
+    progName = progName.exceptLast(3);
+    progName = progName.trim("/\t\n\r");
+    cstring parserFile = progName + "_parser.c";
+    cstring postParserFile = progName + "_post_parser.c";
+    cstring headerFile = progName + "_parser.h";
     if (!options.cFile.isNullOrEmpty()) {
-        auto cstream = openFile(options.cFile, false);
-        if (cstream == nullptr) return;
-        if (ebpf_program == nullptr) return;
-        EBPF::CodeBuilder c(target);
-        ebpf_program->emit(&c);
-        *cstream << c.toString();
-        cstream->flush();
+        parserFile = options.cFile + parserFile;
+        postParserFile = options.cFile + postParserFile;
+        headerFile = options.cFile + headerFile;
     }
+    auto cstream = openFile(postParserFile, false);
+    auto pstream = openFile(parserFile, false);
+    auto hstream = openFile(headerFile, false);
+    if (cstream == nullptr) return;
```

Comment:
Other pstream and hstream nullcheck should also be added. And I think we should also emit error before returning.